### DISCUSSION TITLE
Provided environments .sln files

### DISCUSSION THREAD

**Discussion by joigalcar3** [microsoft/AirSim#3366] [Support Q&A] [answered]
Hello, I am carrying out my Master thesis on fault diagnosis with drones. My goal is to automate the collection of data by flying the drone around multiple randomly generated paths within an urban environment, inducing sensor and actuator failures along the drone path. However, this task is very constrained if the environment can not be opened as an UnrealEngine project (pre-compiled binaries) in order to assess the geography for the path generation or modify the environment in order to improve it for the task at hand. I do not know whether it would be possible to obtain those files for research or there is a work-around for this problem.  

Besides that, when running the executable of the City Environment there are no moving cars or pedestrians. Is it only me who has this problem?

Thank you for your help :)!

**Comment by ahmed-elsaharti** [comment, marked as answer]
Hi @joigalcar3 , due to the fact that some (if not all) of the precompiled environments are based on Unreal Engine assets/environments that are paid, I don't think it would be possible for the maintainers to provide us with the actual sln/uproject files. I could be mistaken though.

You could, however, purchase those environments from sources like the unreal engine marketplace (such as the [neighborhood environment](https://www.unrealengine.com/marketplace/en-US/product/modular-neighborhood-pack?sessionInvalidated=true), etc. Some of them are even free) and then [add AirSim to the custom environment you just created](https://microsoft.github.io/AirSim/unreal_custenv/).
Moreover, if obtaining the locations of the assets/buildings is the issue, you could try using the [`simGetObjectPose` API](https://github.com/microsoft/AirSim/blob/master/docs/apis.md#common-apis) to get the centers of the assets (not very effective since it only returns the center of the object) or even use the new [voxel grid feature](https://github.com/microsoft/AirSim/pull/3209) documented [here](https://github.com/microsoft/AirSim/pull/3239) (this method would probably suit your use-case better).
